src/backports/three_way_comparison.h

```diff
@@ -0,0 +1,384 @@
+//
+// Created by kalmbacj on 9/2/25.
+//
+// Copyright 2025, Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+
+#ifndef THREE_WAY_COMPARISON_H
+#define THREE_WAY_COMPARISON_H
+
+#include <cmath>
+#include <tuple>
+#include <type_traits>
+
+#ifdef QLEVER_CPP_17
+
+namespace ql {
```

Comment:
Can you at the beginning of the file give a short introduction over the various macros that can now be used + a link to the file with the tests that aslo function as examples.